### PR TITLE
Include the filename with navto results for partial symbols to disambiguate them.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Followup to https://github.com/dotnet/roslyn/pull/50128

Before:

image

Here, there's no info to tell you how these parts of the partial types differ.  After you now see:

image

By including the file name, it becomes clearer which part each result will take you to.

Note: i also considered a change where we would not even show a particular part *if* that part only contained nested types and nothing else itself.  However, i shied away from that for a particular edge case.  namely where you had just the following file:

```c#
partial class Outer { class Inner { } }
```

(with no other declaration of `partial class Outer` anywhere else).  IN this case if you searched for 'Outer' you would not find it. 

However, that case seems very niche and unlikely.  So i will be making an alternative PR that explores that idea for us to discuss.

### CODE REVIEW

**Comment by genlu** [comment]
Could you add some test here to cover the change in #50140? I don't want to block the merge of the other PR since it will make the review of this one easier.

**Comment by CyrusNajmabadi** [comment]
> Could you add some test here to cover the change in #50140? I don't want to block the merge of the other PR since it will make the review of this one easier.

i think i'm misunderstanding what you're asking for.  asked for clarity in Teams :)

File: src/Features/Core/Portable/NavigateTo/AbstractNavigateToSearchService.SearchResult.cs [hunk 1]
```diff
@@ -44,29 +44,27 @@ private class SearchResult : INavigateToSearchResult
...
             }
 
-            private static string ComputeAdditionalInformation(Document document, DeclaredSymbolInfo declaredSymbolInfo, ImmutableArray<Project> additionalMatchingProjects)
+            private static string ComputeAdditionalInfo(Document document, DeclaredSymbolInfo info, ImmutableArray<Project> additionalMatchingProjects)
             {
                 var projectName = ComputeProjectName(document, additionalMatchingProjects);
-                switch (declaredSymbolInfo.Kind)
+
+                // For partial types, state what file they're in so the user can disambiguate the results.
+                if (info.IsPartial)
```
Comment (genlu) [inline src/Features/Core/Portable/NavigateTo/AbstractNavigateToSearchService.SearchResult.cs:56]: nit: I feel a switch expression here would make the code clearer

File: src/Features/Core/Portable/FeaturesResources.resx [hunk 2]
```diff
@@ -2822,4 +2822,10 @@ Zero-width positive lookbehind assertions are typically used at the beginning of
   <data name="Removal_of_document_not_supported" xml:space="preserve">
     <value>Removal of document not supported</value>
   </data>
+  <data name="in_0_1_2" xml:space="preserve">
```
Comment (genlu) [inline src/Features/Core/Portable/FeaturesResources.resx:2825]: super nit: in_0_1_dash_2

**Review by genlu — APPROVED** [review]
:shipit:

**Pull request merged**